# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xmp/compile.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -urN xmp-2.0.4.orig/src/loaders/it_load.c xmp-2.0.4/src/loaders/it_load.c
--- xmp-2.0.4.orig/src/loaders/it_load.c	2005-03-29 20:16:41.008201432 +0200
+++ xmp-2.0.4/src/loaders/it_load.c	2005-03-29 20:18:19.885169848 +0200
@@ -327,14 +327,14 @@
 
 #define BUILD_ENV(X) { \
 	    fread (&env, 1, sizeof (env), f); \
-	    xxih[i].##X##ei.flg = env.flg & IT_ENV_ON ? XXM_ENV_ON : 0; \
-	    xxih[i].##X##ei.flg |= env.flg & IT_ENV_LOOP ? XXM_ENV_LOOP : 0; \
-	    xxih[i].##X##ei.flg |= env.flg & IT_ENV_SLOOP ? XXM_ENV_SUS : 0; \
-	    xxih[i].##X##ei.npt = env.num; \
-	    xxih[i].##X##ei.sus = env.slb; \
-	    xxih[i].##X##ei.sue = env.sle; \
-	    xxih[i].##X##ei.lps = env.lpb; \
-	    xxih[i].##X##ei.lpe = env.lpe; \
+	    xxih[i]. X##ei.flg = env.flg & IT_ENV_ON ? XXM_ENV_ON : 0; \
+	    xxih[i]. X##ei.flg |= env.flg & IT_ENV_LOOP ? XXM_ENV_LOOP : 0; \
+	    xxih[i]. X##ei.flg |= env.flg & IT_ENV_SLOOP ? XXM_ENV_SUS : 0; \
+	    xxih[i]. X##ei.npt = env.num; \
+	    xxih[i]. X##ei.sus = env.slb; \
+	    xxih[i]. X##ei.sue = env.sle; \
+	    xxih[i]. X##ei.lps = env.lpb; \
+	    xxih[i]. X##ei.lpe = env.lpe; \
 	    if (env.num) xx##X##e[i] = calloc (4, env.num); \
 	    for (j = 0; j < env.num; j++) { \
 		L_ENDIAN16 (env.node[j].x); \
diff -urN xmp-2.0.4.orig/src/xmms/plugin.c xmp-2.0.4/src/xmms/plugin.c
--- xmp-2.0.4.orig/src/xmms/plugin.c	2005-03-29 20:16:40.999202800 +0200
+++ xmp-2.0.4/src/xmms/plugin.c	2005-03-29 20:19:24.391363416 +0200
@@ -228,7 +228,7 @@
 	xmp_cfg.filter = TRUE;
 	xmp_cfg.pan_amplitude = 80;
 
-#define CFGREADINT(x) xmms_cfg_read_int (cfg, "XMP", #x, &xmp_cfg.##x)
+#define CFGREADINT(x) xmms_cfg_read_int (cfg, "XMP", #x, &xmp_cfg. x)
 
 	filename = g_strconcat(g_get_home_dir(), "/.xmms/config", NULL);
 	if ((cfg = xmms_cfg_open_file(filename))) {
@@ -558,7 +558,7 @@
 	gtk_object_set_data(GTK_OBJECT(xmp_conf_window), #w, w);	\
 	gtk_widget_show(w);						\
 	gtk_box_pack_start(GTK_BOX(vbox6), w, TRUE, TRUE, 0);		\
-	if (xmp_cfg.##o == 1)						\
+	if (xmp_cfg. o == 1)						\
 		gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(w), TRUE); \
 }
 
@@ -658,7 +658,7 @@
 	if (!cfg)
 		cfg = xmms_cfg_new();
 
-#define CFGWRITEINT(x) xmms_cfg_write_int (cfg, "XMP", #x, xmp_cfg.##x)
+#define CFGWRITEINT(x) xmms_cfg_write_int (cfg, "XMP", #x, xmp_cfg. x)
 
 	CFGWRITEINT (mixing_freq);
 	CFGWRITEINT (force8bit);
